test/math/kd_tree_test.cc

```diff
@@ -101,23 +100,121 @@ TEST_F(NodeCmpTest, Test) {
   const std::vector<Vector3> second_level{{6, 5, 2}, {7, 3, 6}};
   const std::vector<Vector3> third_level{{2, 1, 4}, {2, 8, 6}, {9, 1, 4}, {8, 7, 3}};
   const std::vector<Vector3> fourth_level{{1, 4, 1}, {6, 2, 8}, {2, 8, 3}, {1, 8, 9},
                                           {7, 1, 1}, {8, 1, 4}, {9, 4, 1}, {7, 8, 7}};
 
+  // Regions for each level of the kdtree.
+  const AxisAlignedBox first_level_region{{-inf, -inf, -inf}, {inf, inf, inf}};
+  const AxisAlignedBox second_level_region_1{{-inf, -inf, -inf}, {7, inf, inf}};
+  const AxisAlignedBox second_level_region_2{{7, -inf, -inf}, {inf, inf, inf}};
+  const AxisAlignedBox third_level_region_1{{-inf, -inf, -inf}, {7, 5, inf}};
+  const AxisAlignedBox third_level_region_2{{-inf, 5, -inf}, {7, inf, inf}};
+  const AxisAlignedBox third_level_region_3{{7, -inf, -inf}, {inf, 3, inf}};
+  const AxisAlignedBox third_level_region_4{{7, 3, -inf}, {inf, inf, inf}};
+  const AxisAlignedBox fourth_level_region_1{{-inf, -inf, -inf}, {7, 5, 4}};
+  const AxisAlignedBox fourth_level_region_2{{-inf, -inf, 4}, {7, 5, inf}};
+  const AxisAlignedBox fourth_level_region_3{{-inf, 5, -inf}, {7, inf, 6}};
+  const AxisAlignedBox fourth_level_region_4{{-inf, 5, 6}, {7, inf, inf}};
+  const AxisAlignedBox fourth_level_region_5{{7, -inf, -inf}, {inf, 3, 4}};
+  const AxisAlignedBox fourth_level_region_6{{7, -inf, 4}, {inf, 3, inf}};
+  const AxisAlignedBox fourth_level_region_7{{7, 3, -inf}, {inf, inf, 3}};
+  const AxisAlignedBox fourth_level_region_8{{7, 3, 3}, {inf, inf, inf}};
+
+  std::deque<details::Node<Vector3, AxisAlignedBox>> nodes;
+  details::Node<Vector3, AxisAlignedBox>* root{nullptr};
+};
+
+TEST_F(InitializeRegionsTest, Test) {
+  root->set_region(std::make_unique<AxisAlignedBox>(Vector3{{-inf, -inf, -inf}}, Vector3{{inf, inf, inf}}));
+  if (root->get_left() != nullptr) details::Initialize3dRegions(true, root->get_left());
+  if (root->get_right() != nullptr) details::Initialize3dRegions(false, root->get_right());
+
+  // Obtain the nodes.
+  const auto second_level_node_1 = root->get_left();
+  const auto second_level_node_2 = root->get_right();
+  const auto third_level_node_1 = second_level_node_1->get_left();
+  const auto third_level_node_2 = second_level_node_1->get_right();
+  const auto third_level_node_3 = second_level_node_2->get_left();
+  const auto third_level_node_4 = second_level_node_2->get_right();
+  const auto fourth_level_node_1 = third_level_node_1->get_left();
+  const auto fourth_level_node_2 = third_level_node_1->get_right();
+  const auto fourth_level_node_3 = third_level_node_2->get_left();
+  const auto fourth_level_node_4 = third_level_node_2->get_right();
+  const auto fourth_level_node_5 = third_level_node_3->get_left();
+  const auto fourth_level_node_6 = third_level_node_3->get_right();
+  const auto fourth_level_node_7 = third_level_node_4->get_left();
+  const auto fourth_level_node_8 = third_level_node_4->get_right();
```

Comment:
Is there a way to actually set the left hand side from specific indices on the right hand side? We should know the ordering because we control the points and even though the test will require more maintenance but it will work to catch regressions as we need it to work.

Reply:
> Sorry, I don't follow your question.

Instead of `const auto second_level_node_1 = root->get_left();` do `const auto second_level_node_1 = &(nodes_[5])`. (note that 5 is just a made up number).

I don't doubt about you computing them. I just want to make sure we are strict on this test to avoid future failures. It can order them in a different way (different algorithm) and if there is min/max corner that matches that is repeated you might end up with a not covered case.

Reply:
Ahh I see your point, let me know if [e1b76aa](https://github.com/maliput/maliput/pull/520/commits/e1b76aa327acebaedb41d28f2b659a82d562a891) is ok.